### DISCUSSION TITLE
missing values on result of register function

### DISCUSSION THREAD

**Discussion by SJongbloets** [react-hook-form/react-hook-form#12349] [Q&A]
Hi 

I'm have a question regarding the output of the register function. 
When i look at the code, it says it's return value is of type ``UseFormRegisterReturn`` but when i look at the ouput i'm missing the optional values on it. (highly likely that i'm missing something in how i currently use it)

**The use-case for me is the following:**
We have a wrapper component, that holds the actual ``<input />`` element that we pass the ``register`` function to. 
This wrapper component, provides us with some styling for a label, the input field and a possible error message, but we'd like to also add an extra class based on if the field is required. 
The way we solve this now, is to pass an extra prop to the wrapper component, but I saw that the ``register`` return function actually has an optional required value. 

The problem i'm currently having, is that no matter what I do, i can't seem to get that required value to be returned by the register function. Is there anyway to get this value back? This would mean that we'd need to pass less properties to the wrapper component that it could already know.

**Version**
7.53.0

**CSB link:**
https://codesandbox.io/p/sandbox/react-hook-form-register-return-rs5x9t

I hope i've described my issue clearly, any help would be appreciated.

Kind regards,

**Comment by brinsleylogic** [comment]
In the same boat. With this:

```
                        <input
                            {...register("name", {
                                required: "Name is required",
                                minLength: {
                                    value: 3,
                                    message:
                                        "Name must be at least 3 characters",
                                },
                            })}
                        />
```

I get this in the DOM:

```html
<input id=":R4opuuunl7:-form-item" aria-describedby=":R4opuuunl7:-form-item-description" aria-invalid="false" name="name">
```

Would expect `required` and `minLength` to exist on the input element.
